src/components/PostDetails.tsx

```diff
@@ -1,117 +1,98 @@
-import React from 'react';
+/* eslint-disable max-len */
+import React, { useContext, useEffect, useState } from 'react';
 import { Loader } from './Loader';
 import { NewCommentForm } from './NewCommentForm';
+import { GlobalContext } from '../AppContext';
 
 export const PostDetails: React.FC = () => {
+  const {
+    selectedUserPosts,
+    postActiveId,
+    commentsForPost,
+    isLoading,
+    error,
+    selectedUser,
+    handleDeleteComment,
+  } = useContext(GlobalContext);
+
+  const [isFormOpened, setIsFormOpened] = useState(false);
+
+  useEffect(() => {
+    setIsFormOpened(false);
+  }, [postActiveId]);
+
+  if (!selectedUser || !postActiveId || !selectedUserPosts) {
+    return null;
+  }
+
+  const currentPost = selectedUserPosts.find(post => post.id === postActiveId) || null;
+
+  function handleWriteComment() {
+    setIsFormOpened(true);
+  }
+
   return (
     <div className="content" data-cy="PostDetails">
       <div className="content" data-cy="PostDetails">
         <div className="block">
           <h2 data-cy="PostTitle">
-            #18: voluptate et itaque vero tempora molestiae
+            {`#${currentPost?.id}: ${currentPost?.title}`}
           </h2>
 
           <p data-cy="PostBody">
-            eveniet quo quis
-            laborum totam consequatur non dolor
-            ut et est repudiandae
-            est voluptatem vel debitis et magnam
+            {currentPost?.body}
           </p>
         </div>
 
         <div className="block">
-          <Loader />
-
-          <div className="notification is-danger" data-cy="CommentsError">
-            Something went wrong
-          </div>
-
-          <p className="title is-4" data-cy="NoCommentsMessage">
-            No comments yet
-          </p>
-
-          <p className="title is-4">Comments:</p>
-
-          <article className="message is-small" data-cy="Comment">
-            <div className="message-header">
-              <a href="mailto:misha@mate.academy" data-cy="CommentAuthor">
-                Misha Hrynko
-              </a>
-              <button
-                data-cy="CommentDelete"
-                type="button"
-                className="delete is-small"
-                aria-label="delete"
-              >
-                delete button
-              </button>
+          {isLoading && <Loader />}
+          {error && (
```

Comment:
```suggestion
          {!!error && (
```
in rendering with &&  need to convert the value to boolean.